### DISCUSSION TITLE
Getting lemmas for all words in sentence

### DISCUSSION THREAD

**Discussion by moskaliukua** [nlp-compromise/nlp_compromise#1136] [Q&A] [answered]
I'm trying to process a sentence, to get base form for each word
 for example: I am going to the store => I be go to the store
e.g. convert all verbs to infintive forms
I already tried a couple of options
```
import nlp from 'compromise';
const sentence = ' I am going to the store'
const doc = nlp(sentence);
doc.normalize({
      verbs: true,
}).out()
```
and 
`doc.verbs().toInfinitive().out()`
 produces the same result:
 **I go to the store**
the other solution I tried is:
```
doc.terms().map((view) => {
          return view
            .normalize({
              verbs: true,
              plurals: true,
            })
            .out();
        }),
```
which produces this result:
**[
  'I',
  'I be going to the store',
  'I be go to the store',
  'to',
  'the',
  'store'
]**
and
```
doc.terms().map((view) => {
          const term = view.json();
          return nlp(term)
            .normalize({
              verbs: true,
            })
            .out();
        }),
```
is also not enough:
**[ 'I ', 'am ', 'go', 'to ', 'the ', 'store' ]**
I look carefully through the docs, it seems that I missing something. Any help would be appreciated.
Is there a better way to achieve what I'm trying to do?

**Comment by spencermountain** [comment, marked as answer]
hi @moskaliukua - i think this is probably what you're looking for -
```js
let doc = nlp(' I am going to the store')
doc.compute('root')
doc.json()
```
https://runkit.com/spencermountain/66b243b9e0d0c900085ae794
you can see the documentation for root-words [here](https://observablehq.com/@spencermountain/compromise-root)
cheers

**Reply by moskaliukua** [reply]
thanks, that's what I'm looking for, thought `is` should be converted to `be`, but it shouldn't be hard to achieve it from this point

**Comment by hchiam** [comment]
Since your goal is "I am going to the store" => "I **_be_** go to the store", it tells me that each word needs to be processed, since "am going" and "am going to" can sometimes be considered as being derived from "to go" in some contexts. 

And since you want "be" instead of "is", here are 2 alternatives: (i noticed that "am" and also "had" were otherwise not turning into their infinitives "be" and "have")

```js
const sentence = 'I am going to the store';
const docs = sentence.split(' ').map(word => nlp(word));
docs.forEach(word => word.verbs().toInfinitive());
const output_1 = docs.map(doc => doc.text()).join(' ');
document.body.querySelector('#output_1').innerText = output_1;
// I be go to the store
```

or:

```js
const sentence = 'I am going to the store';
const doc = nlp(sentence);
const output_2 = doc.terms().map(view => {
  const wordDoc = nlp(view.text());
  wordDoc.verbs().toInfinitive();
  return wordDoc.text();
}).join(' ');
document.body.querySelector('#output_2').innerText = output_2;
// I be go to the store
```

Here's a demo you edit live without login: https://codepen.io/hchiam/pen/NWZgORW?editors=1000

**Reply by moskaliukua** [reply]
hi @hchiam,
thanks for reply,
Tried all solutions, what is interesting:
 the first solution(with multiple docs) is the slowest **15ms**, 
second one - **5ms**, I also found similar one: 
```
doc.match('#Verb').normalize({
      verbs: true,
    });
``` 
not sure if this is intentend to work this way
But solution with `compute('root')` is only **1ms**. so I picked it up and just manually mapped **is** and **had** to infinitive forms
